dali/kernels/tuple_helpers.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TUPLE_HELPERS_H_
+#define DALI_KERNELS_TUPLE_HELPERS_H_
+
+#include <tuple>
+#include <utility>
+
+namespace dali {
+namespace detail {
+
+template <typename... tuples>
+struct tuple_cat_helper;
+
+template <typename t0>
+struct tuple_cat_helper<t0> { using type = t0; };
+
+template <typename t0, typename... tuples>
+struct tuple_cat_helper<t0, tuples...>
+ : tuple_cat_helper<t0, typename tuple_cat_helper<tuples...>::type> {};
+
+template <typename... T1, typename... T2>
+struct tuple_cat_helper<std::tuple<T1...>, std::tuple<T2...>> {
+  using type = std::tuple<T1..., T2...>;
+};
+
+template <typename... tuples>
+using tuple_cat_t = typename tuple_cat_helper<tuples...>::type;
+
+
+
+template <int... indices>
+struct seq {};
+
+template <typename seq, typename... T>
+struct pack_indices_helper;
+
+template <int... indices, typename T0, typename... T>
+struct pack_indices_helper<seq<indices...>, T0, T...>
```

Comment:
Why don't you use sizeof...(T) to calculate the size of pack, and than you can generate sequences up to given number?

Also you can do it in log(N), using something like:
```
template <int... A, int... B>
build<seq<A...>, seq<B...>> : seq<A..., (sizeof...(A)+B)...> {}
```

Reply:
I'll try...